apps/system/js/bluetooth.js

```diff
@@ -225,6 +226,48 @@ var Bluetooth = {
   },
 
   /**
+   * Clean waitForAdapter timeout.
+   *
+   * @private
+   */
+  _clearTimeout: function bt__clearTimeout() {
+    if (this.responseTimeout != null) {
+      clearTimeout(this.responseTimeout);
+      this.responseTimeout = null;
+    }
+  },
+
+  /**
+   * Get adapter from bluetooth through promise interface.
+   * Since BTv1 get onenabled event before onadapteradded event,
+   * We need watch if adapter is retrieved.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  get waitForAdapter() {
+    return new Promise((resolve, reject) => {
+      if (!Object.prototype.watch) {
```

Comment:
moz only API, Implemented since JavaScript 1.2. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/watch

can be replaced by shim with getter/setter
https://gist.github.com/eligrey/384583

But I think its save to use this api and expect BT APIv1 will be deprecated before this API has been deprecated